suse_migration_services/units/product_setup.py

```diff
@@ -58,29 +60,74 @@ def main():
         products_metadata = os.sep.join(
             [root_path, 'etc', 'products.d']
         )
-        baseproduct = os.sep.join(
-            [products_metadata, 'baseproduct']
-        )
-        if os.path.exists(baseproduct):
+        if os.path.exists(products_metadata):
             log.info('Creating backup of Product data')
             Command.run(
                 [
                     'rsync', '-zav', '--delete', products_metadata + os.sep,
                     '/tmp/products.d.backup/'
                 ]
             )
-            log.info('Updating Base Product to be suitable for migration')
-            xml = ElementTree()
-            xml.parse(baseproduct)
+        baseproduct = get_baseproduct(products_metadata)
+        if baseproduct.count != 1:
+            if not baseproduct.count:
+                message = 'There is no baseproduct'
+            else:
+                message = ('Found multiple product definitions '
+                           'without element <flavor>: \n{0}'
+                           .format('\n'.join(baseproduct.prod_filenames)))
+            log.error(message)
+            raise DistMigrationProductSetupException(message)
+
+        log.info('Updating Base Product to be suitable for migration')
+        delete_target_registration(baseproduct.prod_filenames.pop())
+    except Exception as issue:
+        message = 'Base Product update failed with: {0}'.format(issue)
+        log.error(message)
+        raise DistMigrationProductSetupException(message)
+
+
+def get_baseproduct(products_metadata):
+    base_product_type = namedtuple(
+        'base_product_type', ['count', 'prod_filenames']
+    )
+    prod_filenames = glob.glob(
+        os.path.join(products_metadata, '*.prod')
+    )
+    count = 0
+    names = []
+    xml = ElementTree()
+    for prod_filename in prod_filenames:
+        try:
+            xml.parse(prod_filename)
             register_sections = xml.findall('register')
             for register in register_sections:
-                target_sections = register.findall('target')
-                for target in target_sections:
-                    register.remove(target)
-            xml.write(
-                baseproduct, encoding="UTF-8", xml_declaration=True
-            )
+                flavor = register.findall('flavor')
+                if not flavor:
+                    count = count + 1
+                    names.append(prod_filename)
+
+        except Exception:
+            message = ('Could not parse {0} as XML file'.format(prod_filename))
+            log.error(message)
+            pass
```

Comment:
The pass statement in Python is used when a statement is required syntactically but you do not want any command or code to execute. The pass statement is a null operation; nothing happens when it executes. In the block you are using it there is code, so you don't need this statement here, it's superfluous. Same applies to the other code block where you used it.

In addition I don't see a reason for an extra message variable in this case. So the code can be shortened like this

```python
log.error(
    'Could not parse {0} as XML file'.format(prod_filename)
)
```

Reply:
I thought it was not needed (the pass statement) but I understood that from the comments Robert shared about it.